tests/components/rest/test_binary_sensor.py

```diff
@@ -1,267 +1,381 @@
-        )
-        self.binary_sensor.update()
-        assert not self.binary_sensor.available
-
-    def test_update_when_value_changed(self):
-        """Test state gets updated when sensor returns a new status."""
-        self.rest.update = Mock(
-            "rest.RestData.update",
-            side_effect=self.update_side_effect('{ "key": true }'),
-        )
-        self.binary_sensor.update()
-        assert STATE_ON == self.binary_sensor.state
-        assert self.binary_sensor.available
-
-    def test_update_when_failed_request(self):
-        """Test state gets updated when sensor returns a new status."""
-        self.rest.update = Mock(
-            "rest.RestData.update", side_effect=self.update_side_effect(None)
-        )
-        self.binary_sensor.update()
-        assert not self.binary_sensor.available
 
-    def test_update_with_no_template(self):
-        """Test update when there is no value template."""
-        self.rest.update = Mock(
-            "rest.RestData.update", side_effect=self.update_side_effect("true")
-        )
-        self.binary_sensor = rest.RestBinarySensor(
-            self.hass,
-            self.rest,
-            self.name,
-            self.device_class,
-            None,
-            self.force_update,
-            self.resource_template,
+@respx.mock
+async def test_setup_failed_connect(hass):
+    """Test setup when connection error occurs."""
+    respx.get(
+        "http://localhost", content=httpx.RequestError(message="any", request=Mock())
+    )
+    assert await async_setup_component(
+        hass,
+        binary_sensor.DOMAIN,
+        {
+            "binary_sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 0
+
+
+@respx.mock
+async def test_setup_timeout(hass):
+    """Test setup when connection timeout occurs."""
+    respx.get("http://localhost", content=asyncio.TimeoutError())
+    assert await async_setup_component(
+        hass,
+        binary_sensor.DOMAIN,
+        {
+            "binary_sensor": {
+                "platform": "rest",
+                "resource": "localhost",
+                "method": "GET",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 0
+
+
+@respx.mock
+async def test_setup_minimum(hass):
+    """Test setup with minimum configuration."""
+    respx.get("http://localhost", status_code=200)
+    assert await async_setup_component(
+        hass,
+        binary_sensor.DOMAIN,
+        {
+            "binary_sensor": {
+                "platform": "rest",
+                "resource": "http://localhost",
+                "method": "GET",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+    assert len(hass.states.async_all()) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)